lib/http/request.rb

```diff
@@ -222,6 +224,12 @@ def inspect
     #   @return [String]
     def_delegator :@uri, :host
 
+    def check_errors
+      raise(InvalidURIError, "empty uri provided") if @uri.host.nil?
+      raise(UnsupportedMethodError, "unknown method: #{verb}") unless METHODS.include?(@verb)
+      raise(UnsupportedSchemeError, "unknown scheme: #{scheme}") unless SCHEMES.include?(@scheme)
```

Comment:
IMO, all of the above should raise `ArgumentError`. Please make this method private. Furthermore, what are your thoughts on renaming it to `validate!`?